core/lib/zksync_core/src/api_server/tx_sender/mod.rs

```diff
@@ -953,11 +953,11 @@ impl TxSender {
         // still reject them as it's not.
         let protocol_version = ProtocolVersionId::latest();
         let seal_data = SealData::for_transaction(transaction, tx_metrics, protocol_version);
-        if let Some(reason) = self
-            .0
-            .sealer
-            .find_unexecutable_reason(&seal_data, protocol_version)
-        {
+        if let Some(reason) = self.0.sealer.find_unexecutable_reason(
+            &seal_data,
+            tx_metrics.gas_remaining,
```

Comment:
maybe include it into the `seal_data`?

Reply:
Then it would duplicated for `block_seal_data` and `tx_seal_data`. I'll check, maybe it indeed makes sens to include it and have less args in methods definitions